views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,840 @@
+        },
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          preferences: preference,
+        }),
+      })
+      .then(() => {
+        if (type == 'index') {
+          setState((prevState) => {
+            return {
+              ...prevState,
+              preferences: preference,
+              snackTime: true,
+            }
+          })
+        } else {
+          setState((prevState) => {
+            return {
+              ...prevState,
+              preferences: preference,
+            }
+          })
+        }
+      })
+  }
+  const fetchPreferences = (uid) => {
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      })
+      .then((response) => {
+        if (response.status !== 200) {
+          return
+        }
+        return response.json()
+      })
+      .then((response) => {
+        setState((prevState) => {
+          return {
+            ...prevState,
+            preferences: babyProofPreferences(response),
+          }
+        })
+      })
+  }
+  const fetchConfigItem = (uid, _id) => {
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/configs/${_id}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      })
+      .then((response) => {
+        if (response.status !== 200) {
+          return
+        }
+        return response.json()
+      })
+      .then(() => {})
+  }
+  const fetchConfigurations = async (uid) => {
+    try {
+      const response = await window.fetch(
+        `${basePath}/api/v1/users/${uid}/configs`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'same-origin',
+        }
+      )
+      if (response.status !== 200) return
+      return response.json()
+    } catch (error) {
+      console.log(error)
+    }
+  }
+  const updateConfigurations = (configID, ownsConfig) => {
+    if (ownsConfig) {
+      window
+        .fetch(`${basePath}/api/v1/users/${user.uid}/configs`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'same-origin',
+          body: JSON.stringify({
+            remove: configID,
+          }),
+        })
+        .then(() => {
+          return
+        })
+    } else {
+      window
+        .fetch(`${basePath}/api/v1/users/${user.uid}/configs`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          credentials: 'same-origin',
+          body: JSON.stringify({
+            disable: configID,
+          }),
+        })
+        .then(() => {
+          return
+        })
```

Comment:
do we need this?

Reply:
[ee04cc1](https://github.com/AMP-SCZ/dpdash/pull/304/commits/ee04cc1f53156889ea38ceef5de6725ff205f44f)